### PR TITLE
FTS: Add specific stemmer for monolingual datasets

### PR DESCRIPTION
**Pull request by AndreaFrancis**

### CODE REVIEW

**Comment by severo** [comment]
issue is here: https://github.com/huggingface/dataset-viewer/issues/1437

**Review by lhoestq — COMMENTED** [review]
Cool ! I think you just need to add a migration and we can merge

File: services/worker/src/worker/job_runners/split/duckdb_index.py [hunk 1]
```diff
@@ -74,6 +77,35 @@
 INSTALL_AND_LOAD_EXTENSION_COMMAND = "INSTALL 'fts'; LOAD 'fts';"
 SET_EXTENSIONS_DIRECTORY_COMMAND = "SET extension_directory='{directory}';"
 REPO_TYPE = "dataset"
+# Only some languages are supported, see: https://duckdb.org/docs/extensions/full_text_search.html#pragma-create_fts_index
+STEMMER_MAPPING = {
```
[the remark this answers is not in the document]
Comment (AndreaFrancis) [reply services/worker/src/worker/job_runners/split/duckdb_index.py:82]: Done, I added it

File: services/worker/src/worker/dtos.py [hunk 2]
```diff
@@ -236,6 +236,7 @@ class SplitDuckdbIndex(SplitHubFile):
     num_rows: Optional[int]
     num_bytes: Optional[int]
     duckdb_version: str
+    stemmer: Optional[str]
```
[the remark this answers is not in the document]
Comment (AndreaFrancis) [reply services/worker/src/worker/dtos.py:239]: Sure! I added porter as default for the cache records with 'has_fts': True and None for the ones with 'has_fts': False

**Review by lhoestq — APPROVED** [review]
Awesome !

File: services/worker/src/worker/dtos.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (severo) [reply services/worker/src/worker/dtos.py:239]: so `has_fts` is <strike>useless</strike> redundant now, since it equals to `stemmer is not None`, right?

**Review by severo — APPROVED** [review]

File: services/worker/src/worker/dtos.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (AndreaFrancis) [reply services/worker/src/worker/dtos.py:239]: Yes, I will remove it on an incoming PR (Since there is logic around it, I would like to break anything).

**Pull request merged**